### PR TITLE
Implement vertical tagging.

### PR DESCRIPTION
**Pull request by keyurva**
* Implements the vertical tagging logic in `stat_var_hierarchy_generator.py` but not wired into the main runner (`runner.py`).
* Will integrate in the next PR.

### CODE REVIEW

File: simple/stats/data.py [hunk 1]
```diff
@@ -421,3 +421,15 @@ def to_mcf(self) -> str:
...
+
+
+@dataclass
+class VerticalSpec:
+  population_type: str
+  measured_properties: list[str] = field(default_factory=list)
+  verticals: set[str] = field(default_factory=set)
+
+  def from_json(json: dict) -> Self:
+    return VerticalSpec(population_type=json.get("populationType", ""),
```
Comment (pradh) [inline simple/stats/data.py:433]: must be set.  default to Thing?

File: simple/stats/stat_var_hierarchy_generator.py [hunk 2]
```diff
@@ -177,21 +193,54 @@ class StatVarHierarchy:
...
+  if not vertical_svg:
+    vertical_svg = SVG(vertical_svg_id, _capitalize_and_split(vertical))
+    vertical_svg.parent_svg_ids[sc.DEFAULT_CUSTOM_ROOT_SVG_ID] = True
+  return vertical_svg
+
+
+# Returns a dict from population type to SVG.
+def _get_pop_type_svgs(svgs: dict[str, SVG]) -> dict[str, SVG]:
+  poptype2svg: dict[str, SVG] = {}
+  for _, svg in svgs.items():
```
Comment (pradh) [inline simple/stats/stat_var_hierarchy_generator.py:239]: nit:  values()

File: simple/stats/stat_var_hierarchy_generator.py [hunk 3]
```diff
@@ -177,21 +193,54 @@ class StatVarHierarchy:
...
+                      vertical_specs: list[VerticalSpec]) -> dict[str, SVG]:
+  vertical_svgs: dict[str, SVG] = {}
+  for vertical_spec in vertical_specs:
+    pop_type_svg = poptype2svg.get(vertical_spec.population_type)
+    # If no matching pop type svg, skip.
+    if not pop_type_svg:
+      continue
+    # If no vertical spec mprop in svg mprops, skip.
+    if not any(mprop in vertical_spec.measured_properties
+               for mprop in pop_type_svg.measured_properties):
```
Comment (pradh) [inline simple/stats/stat_var_hierarchy_generator.py:207]: can `if not (x.measured_properties.keys() & y.measured_properties.keys()):` work?

**Review by pradh — APPROVED** [review]

File: simple/stats/data.py (hunk 1 above)
Comment (keyurva) [reply simple/stats/data.py]: Done.

File: simple/stats/stat_var_hierarchy_generator.py (hunk 3 above)
Comment (keyurva) [reply simple/stats/stat_var_hierarchy_generator.py]: Nice simplification - it sure does. Done.

File: simple/stats/stat_var_hierarchy_generator.py (hunk 2 above)
Comment (keyurva) [reply simple/stats/stat_var_hierarchy_generator.py]: Done.

**Review by keyurva — COMMENTED** [review]
Thanks for the review.

**Pull request merged**